### ISSUE TITLE
CORS preflight requests broken

### ISSUE THREAD

**Issue by Fusselwurm** [mandarineorg/mandarinets#276]
**Overview**

Preflight requests do not seem to be recognized as anything, they return with an empty HTML document (and a non-standard header containing the allowed methods for the respective endpoints).

**Scenario**

Example app:

```ts
import { MandarineCore, Controller, GET, Cors } from "https://deno.land/x/mandarinets@v2.3.2/mod.ts";

@Controller('/api')
@Cors({ origin: "http://localhost" })
export class MyApi {

    @GET('/hello')
    public handler() {
        return "hello";
    }

}
new MandarineCore().MVC().run();
```

query with:

`curl --request OPTIONS   --url http://localhost:8080/api/hello   --header 'origin: http://localhost' -D -`

**Expected behavior**

response with at least these headers:

```
HTTP/1.1 204 No Content
Access-Control-Allow-Origin: http://localhost
```

**Actual behavior**

```
HTTP/1.1 200 OK
content-length: 0
content-type: text/html
allowed: HEAD, GET
```

**[Environment]**
 - Mandarine **v2.3.2**
 - Deno **1.8.0**
 - Deno v8 **9.0.257.3**
 - Deno typescript **4.2.2**

**[Additional]**

From a bit of preliminary debugging it looks to me that OPTIONS requests do not even hit the Cors middleware.

Interestingly, issuing a request with the allowed method:

`curl --request GET   --url http://localhost:8080/api/hello   --header 'origin: http://localhost' -D -`  will return an `Access-Control-Allow-Origin` response header.

**Comment by Fusselwurm** [comment]
think I got it. PR shortly.

**Comment by andreespirela** [comment]
fixed in #277
